File: src/main/resources/src/app/beans/vacancy/Vacancy.ts
import { Skill } from "../skill/Skill";

export class Vacancy {

    private _title: string;
    private _description: string;
    private _essentialSkills: string;
    private _preferableSkills: string;
    private _jobType: number;
    private _phoneNumber: string;
    private _email: string;
    private _minSalary: number;
    private _maxSalary: number;
    private _additionalInfoAboutSalary: string;
    private _userId : number;

    public static serialize(vacancy : Vacancy) {
        return {
            title : vacancy._title,
            description : vacancy._description,
            jobType : vacancy._jobType,
            phoneNumber : vacancy._phoneNumber,
            email : vacancy._email,
            minSalary : vacancy._minSalary,
            maxSalary : vacancy._maxSalary,
            essentialSkills: vacancy._essentialSkills,
            preferableSkills: vacancy._preferableSkills,
            additionalInfoAboutSalary : vacancy._additionalInfoAboutSalary
        }
    }

    public static deserialize(object : any): Vacancy {
        
        var vacancy = new Vacancy();

        vacancy.title = object.title;
        vacancy.description = object.description;
        vacancy.jobType = object.jobType;
        vacancy.phoneNumber = object.phoneNumber;
        vacancy.email = object.email;
        vacancy.minSalary = object.minSalary;
        vacancy.maxSalary = object.maxSalary;
        vacancy.essentialSkills = vacancy._essentialSkills;
        vacancy.preferableSkills = vacancy._preferableSkills;
        vacancy.additionalInfoAboutSalary = object.additionalInfoAboutSalary;

        return vacancy;
    }

    get userId(): number {
        return this._userId;
    }

    set userId(value: number) {
        this._userId = value;
    }

    get title(): string {
        return this._title;
    }

    set title(value: string) {
        this._title = value;
    }

    get description(): string {
        return this._description;
    }

    set description(value: string) {
        this._description = value;
    }

    get preferableSkills(): string {
        return this._preferableSkills;
    }

    set preferableSkills(value: string) {
        this._preferableSkills = value;
    }

    get essentialSkills(): string {
        return this._essentialSkills;
    }

    set essentialSkills(value: string) {
        this._essentialSkills = value;
    }

    get jobType(): number {
        return this._jobType;
    }

    set jobType(value: number) {
        this._jobType = value;
    }

    get phoneNumber(): string {
        return this._phoneNumber;
    }

    set phoneNumber(value: string) {
        this._phoneNumber = value;
    }

    get email(): string {
        return this._email;
    }

    set email(value: string) {
        this._email = value;
    }

    get minSalary(): number {
        return this._minSalary;
    }

    set minSalary(value: number) {
        this._minSalary = value;
    }

    get maxSalary(): number {
        return this._maxSalary;
    }

    set maxSalary(value: number) {
        this._maxSalary = value;
    }

    get additionalInfoAboutSalary(): string {
        return this._additionalInfoAboutSalary;
    }

    set additionalInfoAboutSalary(value: string) {
        this._additionalInfoAboutSalary = value;
    }
}

File: src/main/java/com/headkeeper/bean/entity/Vacancy.java
package com.headkeeper.bean.entity;

import org.hibernate.annotations.Type;

import javax.persistence.*;
import java.math.BigDecimal;
import java.util.Set;

@Entity
@Table(name = "vacancy", schema = "head_keepers")
public class Vacancy {

    private int id;
    private String title;
    private String description;
    private String essentialSkills;
    private String preferableSkills;
    private String phoneNumber;
    private String email;
    private BigDecimal minSalary;
    private BigDecimal maxSalary;
    private String additionalInfoAboutSalary;
    private boolean isActive;
    private Set<Skill> skills;
    private User user;

    @Id
    @Column(name = "id", nullable = false)
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    @Basic
    @Column(name = "title", nullable = false, length = 90)
    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    @Basic
    @Column(name = "description", nullable = false, length = -1)
    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    @Basic
    @Column(name = "essential_skills", nullable = false)
    public String getEssentialSkills() {
        return essentialSkills;
    }

    public void setEssentialSkills(String essentialSkills) {
        this.essentialSkills = essentialSkills;
    }

    @Basic
    @Column(name = "preferable_skills", nullable = true)
    public String getPreferableSkills() {
        return preferableSkills;
    }

    public void setPreferableSkills(String preferableSkills) {
        this.preferableSkills = preferableSkills;
    }

    @Basic
    @Column(name = "phone_number", nullable = true, length = 20)
    public String getPhoneNumber() {
        return phoneNumber;
    }

    public void setPhoneNumber(String phoneNumber) {
        this.phoneNumber = phoneNumber;
    }

    @Basic
    @Column(name = "email", nullable = true, length = 90)
    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    @Basic
    @Column(name = "min_salary", nullable = false, precision = 4)
    public BigDecimal getMinSalary() {
        return minSalary;
    }

    public void setMinSalary(BigDecimal minSalary) {
        this.minSalary = minSalary;
    }

    @Basic
    @Column(name = "max_salary", nullable = true, precision = 4)
    public BigDecimal getMaxSalary() {
        return maxSalary;
    }

    public void setMaxSalary(BigDecimal maxSalary) {
        this.maxSalary = maxSalary;
    }

    @Basic
    @Column(name = "additional_info_about_salary", nullable = true, length = 255)
    public String getAdditionalInfoAboutSalary() {
        return additionalInfoAboutSalary;
    }

    public void setAdditionalInfoAboutSalary(String additionalInfoAboutSalary) {
        this.additionalInfoAboutSalary = additionalInfoAboutSalary;
    }

    @Basic
    @Column(name = "is_active", nullable = false)
    @Type(type = "org.hibernate.type.NumericBooleanType")
    public boolean getIsActive() {
        return isActive;
    }

    public void setIsActive(boolean isActive) {
        this.isActive = isActive;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Vacancy vacancy = (Vacancy) o;

        if (id != vacancy.id) return false;
        if (essentialSkills != vacancy.essentialSkills) return false;
        if (isActive != vacancy.isActive) return false;
        if (title != null ? !title.equals(vacancy.title) : vacancy.title != null) return false;
        if (description != null ? !description.equals(vacancy.description) : vacancy.description != null) return false;
        if (preferableSkills != null ? !preferableSkills.equals(vacancy.preferableSkills) : vacancy.preferableSkills != null)
            return false;
        if (phoneNumber != null ? !phoneNumber.equals(vacancy.phoneNumber) : vacancy.phoneNumber != null) return false;
        if (email != null ? !email.equals(vacancy.email) : vacancy.email != null) return false;
        if (minSalary != null ? !minSalary.equals(vacancy.minSalary) : vacancy.minSalary != null) return false;
        if (maxSalary != null ? !maxSalary.equals(vacancy.maxSalary) : vacancy.maxSalary != null) return false;
        if (additionalInfoAboutSalary != null ? !additionalInfoAboutSalary.equals(vacancy.additionalInfoAboutSalary) : vacancy.additionalInfoAboutSalary != null)
            return false;

        return true;
    }

    @Override
    public int hashCode() {
        int result = id;
        result = 31 * result + (title != null ? title.hashCode() : 0);
        result = 31 * result + (description != null ? description.hashCode() : 0);
        result = 31 * result + (preferableSkills != null ? preferableSkills.hashCode() : 0);
        result = 31 * result + (phoneNumber != null ? phoneNumber.hashCode() : 0);
        result = 31 * result + (email != null ? email.hashCode() : 0);
        result = 31 * result + (minSalary != null ? minSalary.hashCode() : 0);
        result = 31 * result + (maxSalary != null ? maxSalary.hashCode() : 0);
        result = 31 * result + (additionalInfoAboutSalary != null ? additionalInfoAboutSalary.hashCode() : 0);
        return result;
    }

    @ManyToMany(mappedBy = "vacancies", fetch = FetchType.EAGER)
    public Set<Skill> getSkills() {
        return skills;
    }

    public void setSkills(Set<Skill> skills) {
        this.skills = skills;
    }

    @ManyToOne
    @JoinColumn(name = "user_id", referencedColumnName = "id", nullable = false, insertable = false, updatable = false)
    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }
}